src/backend/distributed/sql/udfs/enable_citus_mx_for_pre_citus11/latest.sql

```diff
@@ -0,0 +1,34 @@
+CREATE OR REPLACE FUNCTION pg_catalog.enable_citus_mx_for_pre_citus11()
+  RETURNS SETOF regclass
+  LANGUAGE plpgsql
+  AS $$
+DECLARE
+        partitioned_table_count_pre_11 int:=0;
+BEGIN
+
+    SELECT 
+      metadata->'partitioned_table_count_pre_11' INTO partitioned_table_count_pre_11 
+    FROM pg_dist_node_metadata;
+
+    IF partitioned_table_count_pre_11 IS NOT NULL AND partitioned_table_count_pre_11 > 0 THEN
+
+     -- first, fix the partitions
+     SELECT pg_catalog.fix_all_partition_shard_index_names();
+
```

Comment:
if the count is not meaningful, should we just use a boolean?

Reply:
makes sense, done